src/rivian/rivian_ble.py

```diff
@@ -0,0 +1,85 @@
+import platform
+import asyncio
+import secrets
+from asyncio import TimeoutError
+from bleak import BleakScanner, BleakClient, BleakError
+from rivian import utils
+
+RIVIAN_PHONE_ID_VEHICLE_ID_UUID = "AA49565A-4D4F-424B-4559-5F5752495445"
+RIVIAN_PNONCE_VNONCE_UUID = "E020A15D-E730-4B2C-908B-51DAF9D41E19"
+RIVIAN_BLE_ACTIVE_ENTRY_UUID = "5249565F-4D4F-424B-4559-5F5752495445"
+RIVIAN_PHONE_KEY_LOCAL_NAME = "Rivian Phone Key"
+
+# Create an asyncio.Event object to signal the arrival of a new notification.
+notification_event = asyncio.Event()
+notification_data = None
+
+def notification_handler(sender, data):
+    global notification_data
+    notification_data = data
+    notification_event.set()
```

Comment:
We shouldn't use global variables. Instead, move this inside of the `pair_phone` method. Also, please add typing.